packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -0,0 +1,186 @@
+// @flow
+import asyncMap from '@xen-orchestra/async-map'
+import deferrable from 'golike-defer'
+
+import { type Xapi } from '../xapi'
+import {
+  safeDateFormat,
+  type SimpleIdPattern,
+  unboxIdsFromPattern,
+} from '../utils'
+
+import { type Executor, type Job } from './jobs'
+import { type Schedule } from './scheduling'
+
+const METADATA_BACKUP_JOB_TYPE = 'metadataBackup'
+
+type Settings = {|
+  retentionXoMetadata?: number,
+  retentionPoolMetadata?: number,
+|}
+
+type MetadataBackupJob = {
+  ...$Exact<Job>,
+  pools?: SimpleIdPattern,
+  remotes: SimpleIdPattern,
+  settings: $Dict<Settings>,
+  type: METADATA_BACKUP_JOB_TYPE,
+  xoMetadata?: boolean,
+}
+
+const METADATA_BACKUP_DIR = 'xo-metadata-backups'
+
+// File structure on remotes:
+//
+// <remote>
+// └─ xo-metadata-backups
+//   └─ <schedule ID>
+//    ├─ xo
+//    │ └─ <YYYYMMDD>T<HHmmss>.json
```

Comment:
Are these metadata files about the backup or the backed up data itself?

Reply:
The backed up data itself.

Reply:
There should be some metadata, like the job ID, schedule ID, timestamp.

Maybe we'll add other metadata later.